app/inpututils.cpp

```diff
@@ -160,6 +160,35 @@ QString InputUtils::formatNumber( const double number, int precision )
   return QString::number( number, 'f', precision );
 }
 
+QString InputUtils::formatDistanceInProjectUnit( const double distanceInMeters, int precision )
+{
+  QgsCoordinateReferenceSystem projectCrs = QgsProject::instance()->crs();
+
+  switch ( projectCrs.mapUnits() )
+  {
+    case Qgis::DistanceUnit::Meters:
+      return QString( "%1 %2" ).arg( QString::number( distanceInMeters, 'f', precision ), "m" );
+    case Qgis::DistanceUnit::Kilometers:
+      return QString( "%1 %2" ).arg( QString::number( distanceInMeters / 1000, 'f', precision ), "km" );
```

Comment:
Whatabout `QgsDistanceArea:convertLengthMeasurement( distance,  projectCrs.mapUnits() );`

(https://github.com/MerginMaps/input/blob/91519b98a9d473155b5663923b14939bd2d9acda/app/inpututils.cpp#L1617C20-L1617C20)

Reply:
also `QgsUnitTypes::encodeUnit` for units